platform/saivpp/vpplib/SwitchStateBaseRif.cpp

```diff
@@ -379,12 +406,66 @@ void SwitchStateBase::vppProcessEvents ()
 		SWSS_LOG_NOTICE("Received port link event for %s state %s",
 				evp->data.intf_status.hwif_name,
 				evp->data.intf_status.link_up ? "UP" : "DOWN");
-	    }
+	    } else if (evp->type == VPP_BFD_STATE_CHANGE) {
+                SWSS_LOG_NOTICE("Received bfd state change event, multihop:%d, "
+                                "sw_idx:%d, state %d",
+                                evp->data.bfd_notif.multihop,
+                                evp->data.bfd_notif.sw_if_index,
+                                evp->data.bfd_notif.state);
+                asyncBfdStateUpdate(&evp->data.bfd_notif);
+            }
 	    vpp_ev_free(evp);
 	}
     }
 }
 
+sai_status_t SwitchStateBase::asyncBfdStateUpdate(vpp_bfd_state_notif_t *bfd_notif)
+{
+    sai_bfd_session_state_t sai_state;
+    vpp_bfd_info_t bfd_info;
+    memset(&bfd_info, 0, sizeof(bfd_info));
+
+    // Convert vpp state to sai state
+    switch(bfd_notif->state) {
+    case VPP_API_BFD_STATE_ADMIN_DOWN:
+        sai_state = SAI_BFD_SESSION_STATE_ADMIN_DOWN;
+        break;
+    case VPP_API_BFD_STATE_DOWN:
+        sai_state = SAI_BFD_SESSION_STATE_DOWN;
+        break;
+    case VPP_API_BFD_STATE_INIT:
+        sai_state = SAI_BFD_SESSION_STATE_INIT;
+        break;
+    case VPP_API_BFD_STATE_UP:
+        sai_state = SAI_BFD_SESSION_STATE_UP;
+        break;
+    default:
+        sai_state = SAI_BFD_SESSION_STATE_DOWN;
+        break;
+    }
+
+    bfd_info.multihop = bfd_notif->multihop;
+    vpp_to_sai_ip_addr(&bfd_info.local_addr, &bfd_notif->local_addr);
+    vpp_to_sai_ip_addr(&bfd_info.peer_addr, &bfd_notif->peer_addr);
+
+    // Find the BFD session
+    auto it = m_bfd_info_map.find(bfd_info);
```

Comment:
std::map is not thread-safe and here we may have concurrent modification to the map from upstream. We need to protect the map using mutex.